dev/com.ibm.ws.jaxrs.2.0.common/src/com/ibm/ws/jaxrs20/injection/InjectionRuntimeContextHelper.java

```diff
@@ -140,43 +140,44 @@ public static Object getBeanCustomizerContext(JaxRsFactoryBeanCustomizer beanCus
             return;
         }
 
```

Comment:
Maybe add a comment here explaining this synchronization?  And shouldn't you maybe do the synchronized(resource) instead at the top of the method?  (ie. couldn't the "isInit()" also be subject to the race condition?)

Reply:
I think we should limit the sync block to the minimum necessary - otherwise we slow down performance.  The sync ensures that we only "initialize" the ProviderInfo class once.  Once it is initialized, we shouldn't need to reset the provider - keep in mind that the provider is already set when the ProviderInfo was created - so we set it here only when there is a CDI (or EJB) proxy to the provider - so that those technologies can manage the lifecycle.